sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -405,6 +405,80 @@ class WhileStatementExec(
   }
 }
 
+/**
+ * Executable node for RepeatStatement.
+ * @param condition Executable node for the condition.
+ * @param body Executable node for the body.
```

Comment:
Is there any contract/restriction for the `body`?

Reply:
Except for the type, not really. The body can be anything which extends `CompoundBodyExec`